### DISCUSSION TITLE
Output format in YAML metadata

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [jgm/pandoc#9754] [Ideas]
Both Quarto with the [`format:` metadata field](https://quarto.org/docs/output-formats/all-formats.html) and R Markdown with [the `output` field](https://bookdown.org/yihui/rmarkdown/output-formats.html) support specifying output file format in YAML metadata header.

Pandoc seems to have [no such metadata variable](https://pandoc.org/chunkedhtml-demo/6.2-variables.html).

I think this feature would be useful.

See also [this StackOverflow question](https://stackoverflow.com/questions/73667857/select-output-writer-for-pandoc-in-the-yaml-metablock)

**Comment by denismaier** [comment]
I must admit I don't fully understand whether this would really be that useful as this would mean you'd loose one core feature of pandoc: the ability to convert a file to multiple output formats. That said, I guess the easiest solution is to use default files as suggested in the linked thread, or to use a makefile. You might also want to have a look at cli-wrappers/build tools such as the ones listed [here](https://pandoc.org/extras.html#cli-wrappers).

**Reply by jgm** [reply]
This has been discussed extensively in various issues on the issue tracker.  My view is that we should not mix metadata with processing instructions.  Defaults files can be used for the latter.  (And personally I use simple Makefiles to automate repetitive builds.)
